### PR TITLE
Remove view layer from core/network

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

The view layer ended up inside core/network and by view layer, the CLI
output. This would have locked stepped us into a scenario where it would
make it harder (over time) to change the API without changing the CLI.

This fixes that and also fixes an issue where remove wasn't correctly
showing the warnings when NOT using force.

## QA steps

Ensure that the tests pass, but the following steps can be used:

https://github.com/juju/juju/pull/11183

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/3I1QrDuU/2441-fix-the-cli-return-types-for-spaces

File: core/network/space.go [hunk 1]
```diff
@@ -26,11 +26,11 @@ const (
 // ShowSpace represents space information output by the CLI client.
 type ShowSpace struct {
```
Comment (manadart) [inline core/network/space.go]: Was only ever used as a display type. Remove the whole thing.

File: core/network/space.go [hunk 2]
```diff
@@ -26,11 +26,11 @@ const (
...
+	Space SpaceInfo
 	// Application names which are bound to a given space.
-	Applications []string `json:"applications" yaml:"applications"`
+	Applications []string
 	// MachineCount is the number of machines connected to a given space.
-	MachineCount int `json:"machine-count" yaml:"machine-count"`
+	MachineCount int
 }
 
 // RemoveSpace represents space information why a space could not be removed.
```
Comment (manadart) [inline core/network/space.go]: This one can go too.

File: api/spaces/spaces_test.go [hunk 3]
```diff
@@ -234,10 +229,12 @@ func (s *spacesSuite) TestRenameSpaceError(c *gc.C) {
 			Code:    "500",
 		},
 	}}}
-	args := params.RenameSpacesParams{SpacesRenames: []params.RenameSpaceParams{{
-		FromSpaceTag: names.NewSpaceTag(from).String(),
-		ToSpaceTag:   names.NewSpaceTag(to).String(),
-	}}}
+	args := params.RenameSpacesParams{
+		SpacesRenames: []params.RenameSpaceParams{{
```
Comment (manadart) [inline api/spaces/spaces_test.go]: I'd prefer `Changes` over `SpacesRenames`; gives me a tick to read.

File: cmd/juju/space/space.go [hunk 4]
```diff
@@ -235,3 +235,108 @@ func (c *SpaceCommandBase) RunWithSpaceAPI(ctx *cmd.Context, toRun RunOnSpaceAPI
 	defer api.Close()
 	return toRun(api, ctx)
 }
+
+// SubnetInfo is a source-agnostic representation of a subnet.
+// It may originate from state, or from a provider.
+type SubnetInfo struct {
```
Comment (manadart) [inline cmd/juju/space/space.go]: Now that we have mirrored this type for display, is it worth adding an equality test between the core type and the display type that will break with a "you didn't consider this" message if they diverge?

Comment (SimonRichardson) [reply cmd/juju/space/space.go]: Well, the idea is that they're locked stepped at the moment, but that shouldn't always be the case. This what this PR is addressing. They just happen to be the same at the moment.

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**